flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/src/main/java/org/apache/flink/cdc/connectors/mongodb/source/reader/fetch/MongoDBStreamFetchTask.java

```diff
@@ -236,7 +250,7 @@ private MongoChangeStreamCursor<BsonDocument> openChangeStreamCursor(
         BsonDocument resumeToken = offset.getResumeToken();
         BsonTimestamp timestamp = offset.getTimestamp();
 
-        if (resumeToken != null) {
+        if (resumeToken != null && !resumeTokenExpired) {
```

Comment:
It should be noted that `startAtOperationTime` is supported starting from MongoDB version 4.0. 

The prerequisite for executing the following code block is that the `resumeToken` does not exist, which occurs in two scenarios: 
1. a specified timestamp is used as the starting offset.
2. prior to version 4.0.7, when a collection has not been updated for a long time and the `postResumeToken` cannot be obtained, the current timestamp is used as the starting offset.

In the `else` branch of the code below, if we cannot start the Change Stream using `startAtOperationTime ` , we fallback to starting the Change Stream from the latest offset. 
Before MongoDB 4.0, we may lose data during the snapshot changes, but the likelihood of losing data during the snapshot is relatively low: if a collection is highly likely to change, we can obtain the `resumeToken` at its starting position; if a collection has not changed for a long time and we cannot obtain the `resumeToken` and `postResumeToken`, the likelihood of data loss during the snapshot is very low.

However, when we encounter an invalid `resumeToken` during runtime or when recovering from a checkpoint, there is a high possibility of data loss. I have reconsidered this issue, and if we cannot recover the change stream from the `resumeToken`, it is highly likely that we cannot recover it through `startAtOperationTime` either, as they correspond to the same position in the `oplog`.

In `mongo-kafka`, there is an explicit configuration called "tolerant-errors" provided to handle interruptions in the change stream. But in some scenarios, consistency requirements may outweigh availability, we should throw an exception to let user re-run the task from beginning.

```java
       else {
            if (supportsStartAtOperationTime) {
                LOG.info("Open the change stream at the timestamp: {}", timestamp);
                changeStreamIterable.startAtOperationTime(timestamp);
            } else {
                LOG.warn("Open the change stream of the latest offset");
            }
        }
```